[ao][sparsity][fx] make quant prepare -> sparse prepare compose

HDCharles opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #82204
* #81994
* #81993
* __->__ #81992

Summary: sparse_prepare automatically composes with quantized prepare
even in cases with fusion. However, the convert step needed to be updated to handle parametrized
modules.

Test Plan: python test/test_ao_sparsity.py TestFxComposability

Reviewers:

Subscribers:

Tasks:

Tags:

HDCharles committed:
[ao][sparsity][fx] make quant prepare -> sparse prepare compose

Summary: sparse_prepare automatically composes with quantized prepare
even in cases with fusion. However, the convert step needed to be updated to handle parametrized
modules.

Test Plan: python test/test_ao_sparsity.py TestFxComposability

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity][fx] make quant prepare -> sparse prepare compose"


Summary: sparse_prepare automatically composes with quantized prepare
even in cases with fusion. However, the convert step needed to be updated to handle parametrized
modules.

Test Plan: python test/test_ao_sparsity.py TestFxComposability

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on test/ao/sparsity/test_composability.py:
```diff
@@ -312,5 +311,68 @@ def test_qat_prep_before_s_prep(self):
+            .set_module_name("5", qconfig)
+
+
+        mod = prepare_fx(mod, qconfig_mapping, (example))
```
nit: example_inputs is a Tuple[Any, ...], so maybe `(example)` --> `(example,)`

HDCharles replied:
thanks, will fix

jerryzh168 reviewed (approved):
looks good

jerryzh168 commented on test/ao/sparsity/test_composability.py:
```diff
@@ -312,5 +311,68 @@ def test_qat_prep_before_s_prep(self):
+    return False
+
+class TestFxComposability(TestCase):
+    def test_q_prep_fx_before_s_prep(self):
```
can we add some comments for this test describing what this is trying to test

HDCharles replied:
yeah, will add more details to all the tests

HDCharles committed:
Update on "[ao][sparsity][fx] make quant prepare -> sparse prepare compose"


Summary: sparse_prepare automatically composes with quantized prepare
even in cases with fusion. However, the convert step needed to be updated to handle parametrized
modules.

Test Plan: python test/test_ao_sparsity.py TestFxComposability

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles committed:
Update on "[ao][sparsity][fx] make quant prepare -> sparse prepare compose"


Summary: sparse_prepare automatically composes with quantized prepare
even in cases with fusion. However, the convert step needed to be updated to handle parametrized
modules.

Test Plan: python test/test_ao_sparsity.py TestFxComposability

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

HDCharles commented:
@pytorchmergebot merge -g
